ark/utils/spatial_analysis_utils.py

```diff
@@ -435,39 +437,82 @@ def compute_neighbor_counts(current_fov_neighborhood_data, dist_matrix, distlim,
     cell_dist_mat_bin = np.zeros(cell_dist_mat.shape)
     cell_dist_mat_bin[cell_dist_mat < distlim] = 1
 
-    # default is that cell counts itself as a matrix
+    # remove cell as it's own neighbor
     if not self_neighbor:
         cell_dist_mat_bin[cell_dist_mat == 0] = 0
 
     # get num_neighbors for freqs
     num_neighbors = np.sum(cell_dist_mat_bin, axis=0)
 
     # create the 'phenotype has cell?' matrix, excluding non cell-label rows
-    pheno_has_cell = pd.get_dummies(current_fov_neighborhood_data.iloc[:, 2]).to_numpy().T
+    pheno_has_cell_pd = pd.get_dummies(current_fov_neighborhood_data.loc[:, cluster_name_col])
+    pheno_names_from_tab = pheno_has_cell_pd.columns.values
+    pheno_has_cell = pheno_has_cell_pd.to_numpy().T
 
     # dot binarized 'is neighbor?' matrix with pheno_has_cell to get counts
     counts = pheno_has_cell.dot(cell_dist_mat_bin).T
+    counts_pd = pd.DataFrame(counts, columns=pheno_names_from_tab)
+    counts_pd = counts_pd.set_index(current_fov_neighborhood_data.index.copy())
```

Comment:
Is this for code readability or is fixing a bug?

Reply:
Fixing a bug. Setting the columns is important - not all FOVs have the same cell types. This was the problem before that was leading to the wrong results. Setting the index is needed to merge the tables for all FOVs.